compiler/rustc_resolve/src/macros.rs

```diff
@@ -311,6 +316,12 @@ impl<'a> ResolverExpand for Resolver<'a> {
         Ok(ext)
     }
 
+    fn record_macro_rule_usage(&mut self, id: NodeId, rule_i: usize) {
+        if let Some(did) = self.opt_local_def_id(id) {
```

Comment:
```suggestion
        if let def_id = self.local_def_id(node_id) {
```
The def-id must exist if `DUMMY_NODE_ID`s were filtered away previously.